src/components/SQLResults/SQLResultsTable.tsx

```diff
@@ -268,70 +272,74 @@ function SQLResultsTable({ result }: Params) {
     return renderErrorTable(result);
   }
 
-  let columns;
-  let data: object[];
+  let output:
+    | DataTableColumnData<JSONData>
+    | DataTableColumnData<CSVData>
+    | DataTableColumnData<TableViewData>
+    | DataTableColumnData<NoColumnsData>;
 
   if (tableResultsFormat == 'json') {
-    [columns, data] = asJson(result);
+    output = asJson(result);
   } else if (tableResultsFormat == 'csv') {
-    [columns, data] = asCSV(result);
+    output = asCSV(result);
   } else {
-    [columns, data] = asTable(result);
+    output = asTable(result);
   }
 
-  if (columns?.length == 0) {
-    columns = [
-      {
-        title: () => <span className="font-bold">result</span>,
-        key: 'result',
-        dataIndex: 'result',
-        width: '100%',
-        ellipsis: true,
-        className: '',
-      },
-    ];
-    data = [
-      {
-        key: 'OK',
-        result: 'OK',
-      },
-    ];
+  if (output.columns.length == 0) {
+    output = {
+      columns: [
+        {
+          header: () => <span className="font-bold">result</span>,
+          accessorKey: 'result',
+        },
+      ],
+      data: [
+        {
+          result: 'OK',
+        },
+      ],
+    } satisfies DataTableColumnData<NoColumnsData>;
   }
 
   return (
-    <Table
-      columns={columns}
-      dataSource={data}
-      pagination={{ defaultPageSize: 20, position: ['bottomRight'] }}
-      showHeader
-      size="small"
-      sticky
-      scroll={{
-        x: COLUMN_SIZE,
-      }}
-      className="min-h-[50px]"
-      title={() => (
-        <div className="flex h-8 flex-row items-center gap-2">
-          <div className="flex items-center gap-2 border-e pr-2 text-xs font-bold">
-            <Chip className="mr-1.5 bg-green-600 text-white">OK</Chip>
-            {`${result.rowcount} rows, ${(Math.round(result?.duration) / 1000).toFixed(3)} seconds`}
+    <div className="flex h-full w-full flex-col overflow-hidden">
+      <DataTable
+        {...(output as DataTableColumnData<unknown>)}
```

Comment:
Just a note on this, I put the `unknown` here because it's impossible to know the correct type without having a different data table for each format (JSON, CSV, ...) but this does not make any sense at all. 

Typescript is mixing `columns` and `data` types even if that it's not possible (e.g.: JSON column def with CSV data).